tests/EdFi.Security.DataAccess.UnitTests/Repositories/SecurityRepositoryTests.cs

```diff
@@ -165,6 +166,26 @@ public void GetResourceClaimLineage_ShouldReturnEmptyListForInvalidResourceClaim
         // Assert
         lineage.ShouldBeEmpty();
     }
+    
+    [Test]
+    public void GetResourceClaimLineage_ThrowsException_WhenResourceClaimLineageContainsCycle()
+    {
+        // Arrange
+        var resourceClaimA = new ResourceClaim { ClaimName = "uri-A", ResourceClaimId = 1, ParentResourceClaim = null, ParentResourceClaimId = null};
+        var resourceClaimB = new ResourceClaim { ClaimName = "uri-B", ResourceClaimId = 2, ParentResourceClaim = resourceClaimA, ParentResourceClaimId = 1};
+        var resourceClaimC = new ResourceClaim { ClaimName = "uri-C", ResourceClaimId = 3, ParentResourceClaim = resourceClaimB, ParentResourceClaimId = 2};
+        var resourceClaimD = new ResourceClaim { ClaimName = "uri-D", ResourceClaimId = 4, ParentResourceClaim = resourceClaimB, ParentResourceClaimId = 3};
+        
+        resourceClaimA.ParentResourceClaimId = resourceClaimC.ResourceClaimId;
+        resourceClaimA.ParentResourceClaim = resourceClaimC;
+        
+        A.CallTo(() => _securityTableGateway.GetResourceClaims())
+            .Returns(new List<ResourceClaim> { resourceClaimA, resourceClaimB, resourceClaimC });
+
+        // Act & Assert
+        Should.Throw<InvalidOperationException>(() => _securityRepository.GetResourceClaimLineage("uri-C"))
```

Comment:
To include the effects of the `SkipWhile` logic, I think we need to be requesting `uri-D` here, and the `securityTableGateway` mock a couple lines above should be returning all 4 resource claims.